### PR TITLE
Names of color conversion modes have changed / opencv4 requires additional dependency handling

### PR DESCRIPTION
**Pull request by moooeeeep**
The old name will no longer work.

I believe the old versions are deprecated since version 2 and fail to compile with version 3 or 4.

### CODE REVIEW

**Comment by skasperski** [comment]
It's much easier to just add:
`DEPS_CMAKE OpenCV`
to the rock_library macro.

Like here: https://github.com/rock-drivers/drivers-camera_aravis/pull/9/files

**Comment by moooeeeep** [comment]
This is what I had at first, but then I figured out that [the associated orogen package](https://github.com/rock-planning/planning-orogen-exploration) wouldn't build anymore because it doesn't find the opencv headers.

Probably there are other ways to deal with that. Maybe cleaning up the Planner.hpp header might help already (move the opencv to the cpp file).

**Comment by moooeeeep** [comment]
@2maz Do you have an opinion on this? Can you merge this?

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,14 @@
+
+# hack to account for opencv renaming their pkgconfig file with version 4
+find_package(OpenCV REQUIRED)
+if (${OpenCV_VERSION_MAJOR} GREATER 3)
+    set(OPENCV_PKGCONFIG_DEP opencv4)
```
Comment (2maz) [inline src/CMakeLists.txt]: How about using opencv${OpenCV_VERSION_MAJOR}?

Comment (skasperski) [reply src/CMakeLists.txt]: So you're assuming that from now on the OpenCV-devs will follow some logic in naming their pkg-config dependency. ;)

Comment (2maz) [reply src/CMakeLists.txt]: ... and assuming that a pc file will still be provided in the future

**Comment by moooeeeep** [comment]
Found a better way to deal with the opencv dependency. If we remove it from the header it doesn't cause problems down the line.

File: src/CMakeLists.txt (hunk 1 above)
Comment (moooeeeep) [reply src/CMakeLists.txt]: It needs to be future-proof only until we / or [the rock package set](https://github.com/rock-core/rock-package_set/) excludes slam/envire and everything that depends on it, I guess.

**Comment by 2maz** [comment]
@moooeeeep looks like you are removing the function from the public API, so you should consider a deprecation warning

**Comment by moooeeeep** [comment]
I think this should be safe, as this was a private function (and thus not part of the public API) anyways.

**Review by 2maz — APPROVED** [review]

**Pull request merged**